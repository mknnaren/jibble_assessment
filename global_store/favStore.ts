import { find, findIndex } from 'lodash';
import { computed, ref } from "@nuxtjs/composition-api";

const favList:any = ref([])

function addFav(data:{imdbID: string, Title: string, favourite: boolean, Year: string }) { 
    if (!!!find(favList.value, { imdbID: data.imdbID })){
        favList.value.push(data);
    }
}
function removeFav(data: { imdbID: string, Title: string, favourite: boolean, Year: string }) {
    console.log(data);
    console.log(findIndex(favList.value, { imdbID: data.imdbID }));
    let index = findIndex(favList.value, { imdbID: data.imdbID });
    if (index > -1) {
        favList.value.splice(index, 1);
    }
}
function clearAllFav() {
    favList.value.splice(0, favList.value.length);
}
const getFavList = computed(() => favList.value)

export default {
    addFav,
    removeFav,
    clearAllFav,
    getFavList
} 