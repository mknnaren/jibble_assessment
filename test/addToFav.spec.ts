import { mount, shallowMount, createLocalVue  } from '@vue/test-utils'
import addToFavComp from '@/components/movie/add_to_fav.vue';
import  favStore  from '~/global_store/favStore'
import Vuetify from 'vuetify';
import Vue from 'vue'
Vue.use(Vuetify)

describe('addToFavComp', () => {
  let localVue = createLocalVue()
  let vuetify = new Vuetify()

  test('mounts properly', () => {
    const wrapper = mount(addToFavComp, {localVue, vuetify,
      propsData: {
        movieItem: {
          imdbID: "abc123", 
          Title: "Godfather", 
          favourite: true, 
          Year: "1972" 
        }
      }})
    expect(wrapper.vm).toBeTruthy()
  })
  
  test('remove from fav click triggers properly', async () => {
    const wrapper = mount(addToFavComp, {localVue, vuetify,
      propsData: {
        movieItem: {
          imdbID: "abc123", 
          Title: "Godfather", 
          favourite: true, 
          Year: "1972" 
        }
      }})
    await wrapper.find('div#favStarDiv').trigger('click')
    // expect(wrapper.find(".fav-true")).toBeTruthy()
    expect(wrapper.emitted()["fav-update-table"]).toBeTruthy()
    // expect(wrapper.find('div#favStarDiv').exists()).toBe(true)
  })
  test('add to fav click triggers properly', () => {
    const wrapper = mount(addToFavComp, {localVue, vuetify,
      propsData: {
        movieItem: {
          imdbID: "abc123", 
          Title: "Godfather", 
          favourite: false, 
          Year: "1972" 
        }
      }})
    
    wrapper.find('div#favStarDiv').trigger('click')
    // expect(wrapper.find(".fav-true")).toBeTruthy()
    // expect(wrapper.props().movieItem.favourite).toBe('true');
    expect(wrapper.emitted()["fav-update-table"]).toBeTruthy()
    // expect(wrapper.find('div#favStarDiv').exists()).toBe(true)
  })

  test('renders properly', () => {
    const wrapper = mount(addToFavComp,  {localVue, vuetify, propsData: {
      movieItem: {
        imdbID: "abc123", 
        Title: "Godfather", 
        favourite: false, 
        Year: "1972" 
      }
    }})
    expect(wrapper.html()).toMatchSnapshot()
  })
})
