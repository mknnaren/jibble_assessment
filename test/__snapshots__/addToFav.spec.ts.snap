// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addToFavComp renders properly 1`] = `"<div id=\\"favStarDiv\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate fav-false mdi mdi-star-outline theme--light grey--text text--lighten-1\\"></i></div>"`;
